src/components/Layouts/index.js

```diff
@@ -45,6 +47,28 @@ class Layout extends React.Component {
     });
   }
 
+  handleSearchClick = (isSearchOpen) => {
+    this.setState({
+      [isSearchOpen]: !this.state.isSearchOpen
+    }, this.handleSearchEventListener);
+  }
+
+  handleSearchEventListener = () => {
+    if(this.state.isSearchOpen) {
+      document.body.addEventListener('click',this.handleCloseSearchClick);
+    } else {
+      document.body.removeEventListener('click',this.handleCloseSearchClick);
```

Comment:
Do we need to remove the eventListener?

Reply:
Weighing in - I think code like this is important to avoid event listener leak.

Reply:
Yea I was removing the event listener as so it wasn't going off in the background on ever click outside. I will go over the component today and make revisions thanks for the input guys.

Reply:
Makes sense!